File: server/bcephal-grid/src/main/java/com/moriset/bcephal/grid/domain/GrilleBrowserData.java
package com.moriset.bcephal.grid.domain;

import com.moriset.bcephal.domain.BrowserData;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper=false)
public class GrilleBrowserData extends BrowserData {

	private String status;
	private boolean editable;
	private Boolean published;
	
	public GrilleBrowserData(Grille grid) {
		super(grid);
		status = grid.getStatus().name();
		editable = grid.isEditable();
		published = grid.getPublished();
	}
	
	public GrilleBrowserData(MaterializedGrid grid) {
		super(grid);
		editable = grid.isEditable();
		published = grid.isPublished();
	}
}

File: client/bcephal-client-model/Grids/GrilleBrowserData.cs
using Bcephal.Models.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Grids
{
    public class GrilleBrowserData : BrowserData
    {
        public bool? Published { get; set; }
        public string Status { get; set; }

        public bool Editable { get; set; }

    }
}
